addons/web/static/tests/search/action_menus.test.js

```diff
@@ -0,0 +1,97 @@
+import {
+    contains,
+    defineActions,
+    defineModels,
+    fields,
+    models,
+    mountView,
+    onRpc,
+} from "../web_test_helpers";
+import { expect, test } from "@odoo/hoot";
+import { queryAllTexts } from "@odoo/hoot-dom";
+
+/** Foo is dummy model to test `action.report` with domain of its field `value`. **/
+class Foo extends models.Model {
+    _name = "foo";
+
+    value = fields.Boolean();
+
+    _records = [
+        {
+            id: 1,
+            value: true,
+        },
+    ];
+}
+
+defineModels([Foo]);
+
+defineActions([
+    {
+        id: 1,
+        name: "Some Report always visible",
+        report_name: "report_always_visible",
+        report_type: "qweb-pdf",
+        type: "ir.actions.report",
+        model: "foo",
+        binding_model_id: "model_foo",
+        binding_type: "report",
+    },
+    {
+        id: 2,
+        name: "Some Report with satisfied domain",
+        report_name: "report_with_satisfied_domain",
+        report_type: "qweb-pdf",
+        type: "ir.actions.report",
+        model: "foo",
+        binding_model_id: "model_foo",
+        binding_type: "report",
+        domain: [["value", "=", "True"]],
+    },
+    {
+        id: 3,
+        name: "Some Report with no satisfied domain",
+        report_name: "report_without_satisfied_domain",
+        report_type: "qweb-pdf",
+        type: "ir.actions.report",
+        model: "foo",
+        binding_model_id: "model_foo",
+        binding_type: "report",
+        domain: [["value", "=", "False"]],
+    },
+]);
+
+test("render action menu in list view", async () => {
+    onRpc("has_group", () => true);
+    await mountView({
+        type: "list",
+        resModel: "foo",
+        actionMenus: {
+            action: [],
+            print: [
+                {
+                    key: 55,
+                    name: "Custom Print Action",
+                    type: "ir.actions.action_report",
+                    domain: "",
+                },
+            ],
+        },
+        loadActionMenus: true,
+        arch: /* xml */ `
+              <tree>
+                  <field name="value"/>
+              </tree>
+         `,
+    });
+    // select all records
+    await contains(`thead .o_list_record_selector input`).click();
+    expect(`div.o_control_panel .o_cp_action_menus`).toHaveCount(1);
+    expect(queryAllTexts(`div.o_control_panel .o_cp_action_menus .dropdown-toggle`)).toEqual([
+        "Print",
+        "Actions",
+    ]);
+    await contains(`.o_cp_action_menus .dropdown-toggle:eq(0)`).click();
+    expect(`.o-dropdown--menu .o-dropdown-item`).toHaveCount(1);
+    expect(queryAllTexts(`.o-dropdown--menu .o-dropdown-item`)).toEqual(["Custom Print Action"]);
+});
```

Comment:
a single test is a bit light, I would like to have tests
 - to log the network activity (stepAllNetworkCalls)
 - in list views, to select several records => what happens if an action is only available for some of them, but not all
 - in form views, when we do pager previous/next
 - ...